aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      <text>Name: @person.Name</text>
+  }
+
+You generally use this approach when you want to render HTML that is not surrounded by an HTML tag. Without HTML tags, you get a Razor runtime error:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      Name: @person.Name
+  }
+
+.. _explicit-line-transition-with-label:
+
+Explicit Line Transition with ``@:``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Typically, HTML tags provide a boundary for Razor to transition into C#:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      <text>Name: @person.Name</text>
+  }
+
+To render the Razor above without HTML tags, use the ``@:`` characters:
+
+.. code-block:: none
+  :emphasize-lines: 4
+
+  @for (var i = 0; i < people.Length; i++)
+  {
+      var person = people[i];
+      @:Name: @person.Name
+  }
+
+Without the ``@:`` in the code above, you'd get a Razor run time error.
+
+.. _explicit-email-override-label:
+
+``@()`` explicit expression to override email @ symbol
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Consider the following Razor markup:
+
+.. code-block:: none
+  :emphasize-lines: 5
+
+  @{
+      var joe = new Person("Joe", 33);
+   }
+
+  <p>Age @joe.Age</p>
+
+Predictably, the server renders ``<p>Age 33</p>``. But suppose you need to concatenate the output to get ``Age33`` with no space between "Age" and "33". The following markup:
+
+.. code-block:: none
+  :emphasize-lines: 5
+
+  @{
+      var joe = new Person("Joe", 33);
+   }
+
+  <p>Age@joe.Age</p>
+
+generates:
+
+.. code-block:: none
+
+  <p>Age@joe.Age</p>
+
+Razor is treating ``Age@joe.Age`` as an email alias. In cases like this, create an explicit expression with ``@()``:
+
+.. code-block:: none
+
+ <p>Age@(joe.Age)</p>
+
+Which renders
+
+.. code-block:: none
+
+  <p>Age33</p>
+
+.. _control-structures-razor-label:
+
+Control Structures
+------------------
+
+Control structures are an extension of code blocks. All aspects of code blocks (transitioning to markup, inline C#) also apply to the following structures.
+
+Conditionals ``@if``, ``else if``, ``else`` and ``@switch``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``@if`` family controls when code runs:
+
+.. code-block:: none
+
+  @if (value % 2 == 0)
+  {
+      <p>The value was even</p>
+  }
+
+``else`` and ``else if`` don't require the ``@`` symbol:
+
+.. code-block:: none
+
+ @if (value % 2 == 0)
+ {
+     <p>The value was even</p>
+ }
+ else if (value >= 1337)
+ {
+     <p>The value is large.</p>
+ }
+ else
+ {
+     <p>The value was not large and is odd.</p>
+ }
+
+``@switch``
```

Comment:
Since you already list `@switch` in the previous section heading, I don't think it need to be here. Just add a line of prose:

You can use a switch statement like this: